src/orange/orangeinp/UnitProto.hh

```diff
@@ -0,0 +1,202 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/orangeinp/UnitProto.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "geocel/Types.hh"
+#include "orange/OrangeTypes.hh"
+#include "orange/transform/VariantTransform.hh"
+
+#include "ProtoInterface.hh"
+
+namespace celeritas
+{
+namespace orangeinp
+{
+class Transformed;
+
+namespace detail
+{
+struct CsgUnit;
+}  // namespace detail
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a "unit" universe.
+ *
+ * A "unit" is a region of space (specified by the "boundary" object) that is
+ * divided up into multiple smaller regions:
+ * - A "material" (aka "media" in SCALE) is a single homogeneous CSG object
+ *   filled with a particular material ID. This is equivalent to a leaf
+ *   "physical volume" in a GDML/Geant4 volume hierarchy.
+ * - A "daughter" (aka "hole" in SCALE) is another unit that is transformed and
+ *   placed into this universe.
+ *
+ * Regarding boundary conditions: "Input" is for how the unit is *defined*:
+ *
+ *  ========== ==========================================================
+ *   Input      Description
+ *  ========== ==========================================================
+ *   Implicit   Boundary implicitly truncates interior (KENO)
+ *   Explicit   Interior CSG definition includes boundary (RTK)
+ *  ========== ==========================================================
+ *
+ * Additionally, whether the universe is the top-level *global* universe
+ * affects the construction.
+ *
+ *  ========== ==========================================================
+ *   Global?    Description
+ *  ========== ==========================================================
+ *   No         Boundary is already truncated by higher-level unit
+ *   Yes        Boundary must explicitly be represented as a volume
+ *  ========== ==========================================================
+ *
+ * These result in different z ordering for the exterior:
+ *
+ *  === === ================== ========================================
+ *   G   P   Result             Description
```

Comment:
Why does "P"  indicate boundary condition? Also "X" is not good for "explicit" both in general (in my opinion) but also explicitly in this case because "x" already means "exterior" in z-order

Reply:
Also, does each row correspond to a z-order? If so, could you include the z-order as a column?